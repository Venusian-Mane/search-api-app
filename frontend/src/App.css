.App {
  text-align: center;
  margin: auto;
}

img {
  height: 100px;
}

.title {
  color: #202076;
  font-size: 50px;
  text-decoration: underline;
  font-family: Arial, Helvetica, sans-serif;
}

.header {
  background-color: bisque;
}
.headPara {
  font-size: 18px;
  font-weight: bold;
}
.search {
  background-color: #2c27278e;
  width: 90%;
  margin: auto;
}
.search1 {
  width: 90%;
  margin: auto;
}
.resultCard {
  width: 200px;
  height: 300px;
  float: left;
  margin: 10px;
  background-color: black;
  overflow-y: auto;
}
.addFavButton {
  position: relative;
}
.cardText {
  font-size: 14px;
}
.cardImage {
  width: 100px;
  height: 100px;
  float: left;
}
.countryData {
  column-span: all;
}
.tableHead {
  column-span: 7;
}
.inputClass {
  width: 300px;
}
