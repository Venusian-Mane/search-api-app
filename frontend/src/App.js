import React from "react";
import "./App.css";
import Header from "./Components/Header.js";
import Favourites from "./Components/Favourites.js";
import Search from "./Components/Search.js";
import Results from "./Components/Results";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom"; // Routing modules imported for routing to other commponents

function App() {
  return (
    <Router>
      <div className="App">
        <link
          rel="stylesheet"
          href="https://maxcdn.bootstrapcdn.com/bootstrap/4.2.1/css/bootstrap.min.css"
          integrity="sha384-GJzZqFGwb1QTTN6wy59ffF1BuGJpLSa9DkKMp0DgiMDm4iYMj70gZWKYbI706tWS"
          crossorigin="anonymous"
        />
        <Header />
        <Switch>
          <Route path="/search" exact component={Search} />
          <Route path="/results" exact component={Results} />
          <Route path="/favourites" exact component={Favourites} />
          {/* All important components are routed and have unique links */}
        </Switch>
      </div>
    </Router>
  );
}

export default App;
