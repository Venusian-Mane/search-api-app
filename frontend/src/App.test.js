import React from "react";
import renderer from "react-test-renderer";
import ReactDOM from "react-dom";
import App from "./App";
import expressAsyncAwait from "express-async-await";

// this module is totest whether the app renders without crashing
// makig sure there is some kind of output
it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("creates a snapshot", () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
