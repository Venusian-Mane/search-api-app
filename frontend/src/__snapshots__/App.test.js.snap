// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates a snapshot 1`] = `
<div
  className="App"
>
  <link
    crossorigin="anonymous"
    href="https://maxcdn.bootstrapcdn.com/bootstrap/4.2.1/css/bootstrap.min.css"
    integrity="sha384-GJzZqFGwb1QTTN6wy59ffF1BuGJpLSa9DkKMp0DgiMDm4iYMj70gZWKYbI706tWS"
    rel="stylesheet"
  />
  <div>
    <div
      className="header"
    >
      <h1
        className="title"
      >
        iTunes Search App
      </h1>
      <br />
      <p
        className="headPara"
      >
        Find and list all your favourite content here. From Music and Books, to Movies, Audio Books Podcasts and more.
      </p>
      <img
        className="appleLogo"
        src="icons.png"
      />
    </div>
    <div
      className="nav-bar"
    >
      <nav
        className="navbar navbar-expand navbar-dark bg-dark sticky-top"
      >
        <span
          className="navbar-brand"
        >
          Navigate
        </span>
        <div
          className="mr-auto navbar-nav"
          onKeyDown={[Function]}
        >
          <a
            className="nav-link"
            data-rb-event-key="/home"
            href="/home"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Home
          </a>
          <a
            className="nav-link"
            data-rb-event-key="/search"
            href="/search"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Search
          </a>
          <a
            className="nav-link"
            data-rb-event-key="/favourites"
            href="/favourites"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Favourites
          </a>
          <a
            className="nav-link"
            data-rb-event-key="/results"
            href="/results"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Results
          </a>
        </div>
      </nav>
    </div>
  </div>
</div>
`;
